kodi githash dc3168b fails to build: nfs_create/nfs_creat related

<!--- Please fill out this template to the best of your ability. You can always edit this issue once you have created it. -->
<!--- Read the following link before you create a new problem report: https://kodi.wiki/view/HOW-TO:Submit_a_bug_report  -->
## Bug report
### Describe the bug
Here is a clear and concise description of what the problem is:
<!--- Provide a more detailed introduction to the issue itself, and why you consider it to be a bug -->
<!--- A bug report that is not clear will be closed -->
<!--- Put your text below this line -->
Updated kodi to githash dc3168b by creating a pull request for kodi flatpak beta branch: https://github.com/flathub/tv.kodi.Kodi/pull/677/changes/feab9101dc0347e1d0dd43bfedde6b10155f47f4

Build failed: https://github.com/flathub-infra/vorarbeiter/actions/runs/24240000676/job/70772064095

Related to: https://github.com/xbmc/xbmc/pull/28152

## Expected Behavior
Here is a clear and concise description of what was expected to happen:
<!--- Tell us what should happen -->
<!--- Put your text below this line -->
Compilation was expected to succeed


## Actual Behavior
<!--- Tell us what happens instead -->
<!--- Put your text below this line -->
Compilation failed:
```
FAILED: [code=1] build/filesystem/CMakeFiles/filesystem.dir/NFSFile.cpp.o 
/usr/bin/ccache /run/ccache/bin/c++ -DCMS_NO_REGISTER_KEYWORD -DFFMPEG_VER_SHA=7.0.0 -DFMT_SHARED -DHAS_AIRPLAY -DHAS_AIRTUNES -DHAS_ALSA -DHAS_AVAHI -DHAS_DBUS -DHAS_EGL -DHAS_FILESYSTEM_NFS -DHAS_FILESYSTEM_SMB -DHAS_GBM_BO_MAP -DHAS_GBM_MODIFIERS -DHAS_GL -DHAS_GLX -DHAS_ISO9660PP -DHAS_LIRC -DHAS_MARIADB -DHAS_NEON -DHAS_PIPEWIRE -DHAS_PULSEAUDIO -DHAS_PYTHON -DHAS_UDFREAD -DHAS_WEB_INTERFACE -DHAS_WEB_SERVER -DHAS_ZEROCONF -DHAVE_DRM_MODIFIER_NAME -DHAVE_EGLEXTANGLE -DHAVE_GBM -DHAVE_HDR_OUTPUT_METADATA -DHAVE_LCMS2 -DHAVE_LIBBLUETOOTH -DHAVE_LIBBLURAY -DHAVE_LIBBLURAY_BDJ -DHAVE_LIBCAP -DHAVE_LIBCEC -DHAVE_LIBUDEV -DHAVE_LIBVA -DHAVE_LIBVDPAU -DHAVE_LIBXRANDR -DHAVE_LIBXSLT -DHAVE_NEW_CROSSGUID -DHAVE_WAYLAND -DHAVE_X11 -DNPT_CONFIG_ENABLE_LOGGING -DSPDLOG_COMPILED_LIB -DSPDLOG_FMT_EXTERNAL -D_FILE_OFFSET_BITS=64 -I/run/build/kodi -I/run/build/kodi/xbmc -I/run/build/kodi/xbmc/platform/linux -I/run/build/kodi/xbmc/cores/VideoPlayer -I/run/build/kodi/build -I/run/build/kodi/build/include -I/run/build/kodi/xbmc/platform/posix -isystem /usr/include/fribidi -isystem /usr/include/freetype2 -isystem /usr/include/libpng16 -isystem /usr/include/harfbuzz -isystem /usr/include/glib-2.0 -isystem /usr/lib/aarch64-linux-gnu/glib-2.0/include -isystem /usr/include/uuid -isystem /run/build/kodi/build/cores/RetroPlayer/messages -isystem /usr/include/lzo -isystem /usr/include/taglib -isystem /usr/include/libdrm -isystem /usr/include/dbus-1.0 -isystem /usr/lib/aarch64-linux-gnu/dbus-1.0/include -isystem /usr/include/libxml2 -isystem /app/include/libcec -isystem /usr/include/p11-kit-1 -isystem /usr/include/pipewire-0.3 -isystem /usr/include/spa-0.2 -isystem /usr/include/python3.13 -isystem /app/include/mariadb -isystem /run/build/kodi/lib/libUPnP -isystem /run/build/kodi/lib/libUPnP/Platinum/Source/Core -isystem /run/build/kodi/lib/libUPnP/Platinum/Source/Platinum -isystem /run/build/kodi/lib/libUPnP/Platinum/Source/Devices/MediaConnect -isystem /run/build/kodi/lib/libUPnP/Platinum/Source/Devices/MediaRenderer -isystem /run/build/kodi/lib/libUPnP/Platinum/Source/Devices/MediaServer -isystem /run/build/kodi/lib/libUPnP/Neptune/Source/Core -isystem /run/build/kodi/lib/libUPnP/Neptune/Source/System/Posix -O2 -pipe -g -Wp,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -fexceptions -fstack-protector-strong -grecord-gcc-switches -fasynchronous-unwind-tables -fstack-clash-protection -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -mbranch-protection=bti+pac-ret -Wall -Wdouble-promotion -Wmissing-field-initializers -Wsign-compare -Wextra -Wno-unused-parameter -Wnon-virtual-dtor -O3 -DNDEBUG -std=c++20 -DTARGET_POSIX -DTARGET_LINUX -D_GNU_SOURCE -DHAVE_LINUX_UDMABUF=1 -DHAVE_LINUX_DMA_HEAP=1 -DHAVE_LINUX_DMA_BUF=1 -DHAVE_MKOSTEMP=1 -DHAVE_GETMNTENT_R=1 -DHAVE_LINUX_MEMFD=1 -DHAVE_STATX=1 -D__STDC_CONSTANT_MACROS -D_FILE_OFFSET_BITS=64 -DHAS_POSIX_NETWORK -DHAS_LINUX_NETWORK -DHAS_BUILTIN_SYNC_ADD_AND_FETCH=1 -DHAS_BUILTIN_SYNC_SUB_AND_FETCH=1 -DHAS_BUILTIN_SYNC_VAL_COMPARE_AND_SWAP=1 -DHAVE_INOTIFY=1 -DHAVE_POSIX_FADVISE=1 -DHAVE_LOCALTIME_R=1 -DHAVE_GMTIME_R=1 -DHAVE_INTTYPES_H=1 -DHAS_UPNP=1 -DHAS_OPTICAL_DRIVE -DHAS_CDDA_RIPPER -DBIN_INSTALL_PATH=\"/app/lib/kodi\" -DINSTALL_PATH=\"/app/share/kodi\" -Werror=double-promotion -Werror=missing-field-initializers -Werror=sign-compare -D_REENTRANT -MD -MT build/filesystem/CMakeFiles/filesystem.dir/NFSFile.cpp.o -MF build/filesystem/CMakeFiles/filesystem.dir/NFSFile.cpp.o.d -o build/filesystem/CMakeFiles/filesystem.dir/NFSFile.cpp.o -c /run/build/kodi/xbmc/filesystem/NFSFile.cpp
/run/build/kodi/xbmc/filesystem/NFSFile.cpp: In member function ‘virtual bool XFILE::CNFSFile::OpenForWrite(const CURL&, bool)’:
/run/build/kodi/xbmc/filesystem/NFSFile.cpp:946:11: error: ‘nfs_create’ was not declared in this scope; did you mean ‘nfs_creat’?
  946 |     ret = nfs_create(m_pNfsContext, filename.c_str(), O_RDWR | O_EXCL,
      |           ^~~~~~~~~~
      |           nfs_creat
[341/1887] Building CXX object build/filesystem/CMakeFiles/filesystem.dir/NFSDirectory.cpp.o
```


## Possible Fix
<!--- Not obligatory, but suggest a fix or reason for the bug -->
<!--- Put your text below this line -->



### To Reproduce
Steps to reproduce the behavior:
<!--- Provide a link to a live example, or an unambiguous set of steps to -->
<!--- reproduce this bug. Include code to reproduce, if relevant -->
<!--- Put your text below this line -->
1. 
2.
3.


### Debuglog
<!--- Put your text below this line -->
<!--- A debuglog is always mandatory when creating an issue. Provide one! -->
The debuglog can be found here:



### Screenshots 
Here are some links or screenshots to help explain the problem:
<!--- Put your text below this line -->



## Additional context or screenshots (if appropriate)
Here is some additional context or explanation that might help:
<!--- How has this bug affected you? What were you trying to accomplish? -->
<!--- Put your text below this line -->



### Your Environment
Used Operating system:
<!--- Include as many relevant details about the environment you experienced the bug in -->
<!--- Put your text below this line. Checkboxes can easily be ticked once issue is created -->
 - [ ] Android
 - [ ] iOS
 - [ ] tvOS
 - [x] Linux
 - [ ] macOS
 - [ ] Windows
 - [ ] Windows UWP

 - Operating system version/name:
 - Kodi version:



<!--- End of this issue -->
*note: Once the issue is made we require you to update it with new information or Kodi versions should that be required.
Team Kodi will consider your problem report however, we will not make any promises the problem will be solved.*

--- a/xbmc/filesystem/NFSFile.cpp
+++ b/xbmc/filesystem/NFSFile.cpp
@@ -942,18 +942,12 @@ bool CNFSFile::OpenForWrite(const CURL& url, bool bOverWrite)
   {
     CLog::Log(LOGWARNING, "FileNFS::OpenForWrite() called with overwriting enabled! - {}",
               filename);
-    //create file with proper permissions
-    ret = nfs_create(m_pNfsContext, filename.c_str(), O_RDWR | O_EXCL,
-                     S_IRUSR | S_IWUSR | S_IRGRP | S_IROTH, &m_pFileHandle);
-    //if file was created the file handle isn't valid ... so close it and open later
-    if(ret == 0)
-    {
-      nfs_close(m_pNfsContext,m_pFileHandle);
-      m_pFileHandle = NULL;
-    }
   }
 
-  ret = nfs_open(m_pNfsContext, filename.c_str(), O_RDWR, &m_pFileHandle);
+  // nfs_open2 handles both creation and open atomically;
+  const int flags = bOverWrite ? O_CREAT | O_RDWR | O_EXCL : O_RDWR;
+  const int mode = bOverWrite ? S_IRUSR | S_IWUSR | S_IRGRP | S_IROTH : 0;
+  ret = nfs_open2(m_pNfsContext, filename.c_str(), flags, mode, &m_pFileHandle);
 
   if (ret || m_pFileHandle == NULL)
   {

